src/components/FilesGallery/components/VideoFileView/VideoFileView.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+import {Spin} from '@gravity-ui/uikit';
+
+import {block} from '../../../utils/cn';
+import {FilesGalleryFallbackText} from '../FallbackText/FallbackText';
+
+import {i18n} from './i18n';
+
+import './VideoFileView.scss';
+
+const cnVideoFileView = block('video-file-view');
+
+export type VideoFileViewProps = {
+    className?: string;
+    src: string;
+};
+
+export const VideoFileView = ({className, src}: VideoFileViewProps) => {
+    const [status, setStatus] = React.useState<'loading' | 'playing' | 'error'>('loading');
+    const [errorMessage, setErrorMessage] = React.useState<string>();
+
+    const handleLoad = React.useCallback(() => {
+        setStatus('playing');
+    }, []);
+
+    const handleError = React.useCallback((event: React.SyntheticEvent<HTMLVideoElement>) => {
+        setStatus('error');
+
+        const error = (event.target as HTMLVideoElement).error;
+
+        setErrorMessage(
+            error?.MEDIA_ERR_SRC_NOT_SUPPORTED ? i18n('not-supported-video') : i18n('video-error'),
+        );
+    }, []);
+
+    if (status === 'error') {
+        return <FilesGalleryFallbackText>{errorMessage}</FilesGalleryFallbackText>;
+    }
+
+    return (
+        <React.Fragment>
+            {status === 'loading' && <Spin className={cnVideoFileView('spin')} size="xl" />}
+            {/* eslint-disable-next-line jsx-a11y/media-has-caption */}
+            <video
+                className={cnVideoFileView(null, className)}
+                src={src}
+                controls
+                autoPlay
```

Comment:
![image](https://github.com/user-attachments/assets/b981c8a8-f103-4cd8-9067-a7dff9aa776b)
It's impossible to interact with video after its panel is gone. Hover does not work and click switches the file. Is that how it's supposed to work?
Auto-play can also be quite annoying in such a case.

Reply:
| Auto-play can also be quite annoying in such a case.

This is already user tested scenario. But I can make the autoplay optional.